tests/vali_tests/test_weights_unit.py

```diff
@@ -96,27 +263,88 @@ def test_transform_and_scale_results_defaults(self):
+        sample_gains = [0.0, 0.1, 0.0, 0.1, 0.0, 0.0]
+        sample_losses = [0.0, -0.05, -0.1, -0.1, -0.3, 0.0]
+        sample_n_updates = [ 1, 1, 1, 1, 1, 1 ]
+        sample_open_ms = [ 100, 200, 300, 400, 500, 600 ]
+
+        omega_negative = Scoring.omega_cps(
+            gains=sample_gains,
+            losses=sample_losses,
+            n_updates=sample_n_updates,
+            open_ms=sample_open_ms,
+        )
+
+        # Omega should be less than 1
+        self.assertLessEqual(omega_negative, 1.0)
+        self.assertGreaterEqual(omega_negative, 0.0)
+
+    def test_omega_cps_handcalculation(self):
+        """Test that the omega function works as expected"""
+        sample_gains = [0.0, 0.1, 0.0, 0.1, 0.0, 0.0]
+        sample_losses = [0.0, -0.05, -0.1, -0.1, -0.3, 0.0]
+        sample_n_updates = [ 1, 1, 1, 1, 1, 1 ]
+        sample_open_ms = [ 100, 200, 300, 400, 500, 600 ]
+
+        ## returns - ( 1 + threshold ) -> we're ignoring threshold for internal calculations
+        ## positive returns should be [ 1.1, 1.2, 1.3, 1.4, 1.2 ] -> [ 0.1, 0.2, 0.3, 0.4, 0.2 ]
+        ## negative returns should be [ 0.9, 0.7 ] -> [ -0.1, -0.3 ]
+        hand_computed_omega = sum(sample_gains) / abs(sum(sample_losses))
+
+        ## omega should be [ 1.1 + 1.2 + 1.3 + 1.4 + 1.2 ] / [ 0.9 + 0.7 ]
+        omega = Scoring.omega_cps(
+            gains=sample_gains,
+            losses=sample_losses,
+            n_updates=sample_n_updates,
+            open_ms=sample_open_ms,
+        )
+        self.assertEqual(omega, hand_computed_omega)
+
+    def test_omega_cps(self):
+        """Test inverted sortino function works as expected"""
+        omega_scores = {}
+        for miner, minerledger in self.ledger_dict.items():
+            gains = [ cp.gain for cp in minerledger.cps ]
+            losses = [ cp.loss for cp in minerledger.cps ]
+            n_updates = [ cp.n_updates for cp in minerledger.cps ]
+            open_ms = [ cp.open_ms for cp in minerledger.cps ]
+
+            score = Scoring.omega_cps(
+                gains=gains,
+                losses=losses,
+                n_updates=n_updates,
+                open_ms=open_ms
+            )
+
+            omega_scores[miner] = score
 
-        sample_returns = [ np.log(x) for x in sample_returns ]
-        omega_negative = Scoring.omega(sample_returns)
+        ## the good miner should have the highest score
+        self.assertGreater(omega_scores['good'], omega_scores['bad'])
 
-        # sum above threshold should be 0
-        self.assertEqual(omega_negative, 0.0)
+        ## Good miner should be better than the average miner
+        self.assertGreater(omega_scores['good'], omega_scores['miner0'])
+
+        ## High swings miner will likely be better than the lowswings miner for omega
+        self.assertGreater(omega_scores['highswings'], omega_scores['lowswings'])
+        self.assertGreater(omega_scores['highswingsactive'], omega_scores['highswingsinactive'])
```

Comment:
@jbonilla-tao you can compare the results for 'highswingactive' and 'highswinginactive' for your concerns. If the active miner achieves gains with the same level of time adjusted risk in a new trade, their omega will increase while the sortino will stay the same. This will result in an increased score

Reply:
thanks for testing this